rich/syntax.py

```diff
@@ -374,8 +381,8 @@ def highlight(
         )
         _get_theme_style = self._theme.get_style_for_token
         try:
-            lexer = get_lexer_by_name(
-                self.lexer_name,
+            lexer = self.lexer or get_lexer_by_name(
```

Comment:
Breaking this in to an `if` rather than an expression would allow test coverage to span both branches, and avoid that cast.

Reply:
I kinda prefer the previous version 😊 ideally the if would live outside the try/except, but there's logic using the lexer inside the else statement, so I've kept it there. We still need a cast (or an asser as I've done now) in this case :)